navigation/navigation-compose-material/src/main/java/androidx/navigation/compose/material/BottomSheetNavigator.kt

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.navigation.compose.material
+
+import androidx.compose.foundation.layout.ColumnScope
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.material.ModalBottomSheetState
+import androidx.compose.material.ModalBottomSheetValue
+import androidx.compose.material.rememberModalBottomSheetState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.saveable.rememberSaveableStateHolder
+import androidx.lifecycle.Lifecycle
+import androidx.navigation.FloatingWindow
+import androidx.navigation.NavBackStackEntry
+import androidx.navigation.NavDestination
+import androidx.navigation.NavOptions
+import androidx.navigation.Navigator
+import androidx.navigation.NavigatorState
+import androidx.navigation.compose.material.BottomSheetNavigator.Destination
+
+/**
+ * Create and remember a [BottomSheetNavigator]
+ *
+ * @param sheetState The [ModalBottomSheetState] that the [BottomSheetNavigator] will use to
+ * drive the sheet state
+ */
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+public fun rememberBottomSheetNavigator(
+    sheetState: ModalBottomSheetState =
```

Comment:
I'm still not entirely on board with having this an an input given that we specifically don't want them to set a different `initialValue` or `confirmStateChange`, not to mention wanting to avoid the calls to `show()` and `hide()` externally.

Perhaps we can expose the public properties of `ModalBottomSheetState` as part of `BottomSheetNavigator`'s API surface, thus allowing developers to use the Navigator directly where they were using the state while maintaining the ownership properties we need.

Reply:
Yeah I agree. Done in d0a03a0, would love to get your thoughts about using an extra class to represent that. Adding these to the navigator felt a bit bloaty, but discoverability is worse this way.